src/main/kotlin/org/jetbrains/research/testspark/tools/factories/TestCompilerFactory.kt

```diff
@@ -24,33 +24,41 @@ object TestCompilerFactory {
 
         // TODO add the warning window that for Java we always need the javaHomeDirectoryPath
         return when (language) {
-            SupportedLanguage.Java -> createJavaCompiler(project, libraryPaths, junitLibraryPaths, javaHomeDirectory)
-            SupportedLanguage.Kotlin -> createKotlinCompiler(libraryPaths, junitLibraryPaths)
+            SupportedLanguage.Java -> {
+                val javaSDKHomePath = findJavaSDKHomePath(javaHomeDirectory, project)
+                JavaTestCompiler(libraryPaths, junitLibraryPaths, javaSDKHomePath)
+            }
+            SupportedLanguage.Kotlin -> {
+                val javaSDKHomePath = findJavaSDKHomePath(javaHomeDirectory, project) // Kotlinc relies on java to compile kotlin files.
```

Comment:
**nitpick**: I'd comment on the line above. It is 120 characters long, which is quite long compared to the neighboring lines.

Reply:
You are right, this is something I realised after committing that it would probably be nicer if I placed the comment above the line rather than next to it. Should be fixed now. Thank you! 🙂